### PR TITLE
feat: added support for vpc flow logs

### PR DESCRIPTION
**Pull request by oasisk**

### CODE REVIEW

**Review by bjp232004 — APPROVED** [review]

File: src/service/logs/kinesis_firehose.rs [hunk 1]
```diff
@@ -112,71 +113,70 @@ pub async fn process(
...
+                                Err(_e) => {
+                                    local_val.insert("message".to_owned(), event.message.clone());
+                                }
+                                Ok(message_val) => {
+                                    local_val.insert("message".to_owned(), message_val.clone());
+                                }
                             }
+                        } else {
+                            local_val.insert("message".to_owned(), local_msg.into());
                         }
```
Comment (vvv) [inline src/service/logs/kinesis_firehose.rs:156]:
```suggestion
                        let msg = event.message.as_str().unwrap();
                        let msg = if msg.starts_with('{') && msg.ends_with('}') {
                            json::from_str::<json::Value>(msg)
                                .unwrap_or_else(|_| event.message.clone())
                        } else {
                            msg.into()
                        };
                        local_val.insert("message".to_owned(), msg.into());
```

File: src/service/logs/kinesis_firehose.rs [hunk 2]
```diff
@@ -112,71 +113,70 @@ pub async fn process(
...
+                            Some(v) => match parse_i64_to_timestamp_micros(v) {
+                                Ok(t) => t,
+                                Err(e) => {
+                                    stream_status.status.failed += 1;
+                                    stream_status.status.error = e.to_string();
+                                    continue;
+                                }
+                            },
+                            None => Utc::now().timestamp_micros(),
+                        };
```
Comment (vvv) [inline src/service/logs/kinesis_firehose.rs:170]: Define a helper function
```rust
fn try_parse_timestamp(t: Option<i64>) -> Result<i64, anyhow::Error> {
    t.map_or_else(
        || Ok(Utc::now().timestamp_micros()),
        parse_i64_to_timestamp_micros,
    )
}
```
and use it here ..
```suggestion
                        timestamp = match try_parse_timestamp(event.timestamp) {
                            Ok(t) => t,
                            Err(e) => {
                                stream_status.status.failed += 1;
                                stream_status.status.error = e.to_string();
                                continue;
                            }
                        };```

File: src/service/logs/kinesis_firehose.rs [hunk 3]
```diff
@@ -112,71 +113,70 @@ pub async fn process(
...
-                    let mut value = json::flatten_json_and_format_field(&value);
-                    // get json object
-                    let local_val = value.as_object_mut().unwrap();
-
-                    // handle timestamp
-                    let timestamp = match event.timestamp {
-                        Some(v) => match parse_i64_to_timestamp_micros(v) {
+                } else {
+                    value = json::from_str(&decompressed_data)?;
+                    timestamp = match value
```
Comment (vvv) [inline src/service/logs/kinesis_firehose.rs:174]: .. and here
```rust
                        timestamp = match try_parse_timestamp(event.timestamp) {
                            Ok(t) => t,
                            Err(e) => {
                                stream_status.status.failed += 1;
                                stream_status.status.error = e.to_string();
                                continue;
                            }
                        };
```

File: src/service/logs/kinesis_firehose.rs [hunk 4]
```diff
@@ -325,12 +349,14 @@ pub async fn process(
     }))
 }
 
-fn decode_and_decompress(encoded_data: &str) -> Result<String, Box<dyn std::error::Error>> {
+fn decode_and_decompress(encoded_data: &str) -> Result<(String, bool), Box<dyn std::error::Error>> {
```
Comment (vvv) [inline src/service/logs/kinesis_firehose.rs:352]: Please document the returned `bool`. I don't understand it.

If you apply the suggestion, don't forget to fill the `[XXX-DOCUMENTME]` placeholder.
```suggestion
/// Decode `input` from Base64 representation and decompress the result into a `String`.
/// Returned `bool` signifies whether [XXX-DOCUMENTME].
fn decode_and_decompress(input: &str) -> Result<(String, bool), anyhow::Error> {
```

File: src/service/logs/kinesis_firehose.rs [hunk 5]
```diff
@@ -343,26 +369,24 @@ mod tests {
...
-        let result = decode_and_decompress(encoded_data);
-        assert!(
-            result.is_err(),
-            "Expected an error due to invalid base64 input"
-        );
+    fn test_decode_success() {
+        let encoded_data = "eyJtZXNzYWdlIjoiMiAwNTg2OTQ4NTY0NzYgZW5pLTAzYzBmNWJhNzlhNjZlZjE3IDEwLjMuMTY2LjcxIDEwLjMuMTQxLjIwOSA0NDMgMzg2MzQgNiAxMDMgNDI5MjYgMTY4MDgzODU1NiAxNjgwODM4NTc4IEFDQ0VQVCBPSyJ9Cg==";
+        let expected = "{\"message\":\"2 058694856476 eni-03c0f5ba79a66ef17 10.3.166.71 10.3.141.209 443 38634 6 103 42926 1680838556 1680838578 ACCEPT OK\"}\n";
+        let result = decode_and_decompress(encoded_data).expect("Failed to decode data");
+        assert_eq!(result.0, expected);
```
Comment (vvv) [inline src/service/logs/kinesis_firehose.rs:380]: We never use `result.1`. Why do we need that boolean flag at all?

**Review by vvv — APPROVED** [review]

**Pull request merged**